managedplugin/download.go

```diff
@@ -411,3 +411,23 @@ func WithBinarySuffix(filePath string) string {
 	}
 	return filePath
 }
+
+func FindLatestPluginVersion(ctx context.Context, c *cloudquery_api.ClientWithResponses, ops HubDownloadOptions) (string, error) {
+	if ops.TeamName == "" {
+		return "", fmt.Errorf("team name is required to find the latest plugin version")
+	}
+
+	resp, err := c.ListPluginVersionsWithResponse(ctx, ops.PluginTeam, cloudquery_api.PluginKind(ops.PluginKind), ops.PluginName, &cloudquery_api.ListPluginVersionsParams{})
```

Comment:
`GetPlugin` has a `LatestVersion` field which leads to less data being transferred and leaves the page management/ordering stuff to the backend.

Reply:
`resp, err := c.GetPluginWithResponse(ctx, ops.PluginTeam, cloudquery_api.PluginKind(ops.PluginKind), ops.PluginName)`
Use `resp.JSON200.LatestVersion`

Reply:
I reworked the code based on:
- How it's gonna be used. It didn't work well as a standalone function. I put it in the `Client` that `sync` gets.
- I replaced the API call to use `GetPluginWithResponse` as per feedback.

Now, a sample usage in CLI looks like this:

```
latestVersion, err := destPluginClient.FindLatestPluginVersion(ctx, managedplugin.PluginDestination)
```

And a sample check to see if it works:

```
$ go run . test-connection /Users/mariano.gappa/Code/test-aws-sync/all.yml

Loading spec(s) from /Users/mariano.gappa/Code/test-aws-sync/all.yml
You chose version v27.23.0. Latest version is v27.23.1.
```